Update the rest api documentation for multiclass support (#55)
label-sleuth/label-sleuth.org: docs/docs/dev/api.md

      {
         "category_description": "",
         "category_name": "category1",
         "category_id": "0",
         "color": "amber",
         "deleted": true,
      },
      {
         "category_description": "",
         "category_name": "category2",
         "category_id": "1",
         "color": "amber",
         "deleted": false,
      }
   ]
}
```

:::
::::

## Label Management

Endpoints related to user-provided labels.

### Set element label

Update the label of a selected element. This endpoint either sets or removes the label for a given element.

> - In binary workspaces, the `binary_label` is set to the `category_id`. If `binary_label == 'none'`, the element's label will be removed. Otherwise, the element will be assigned a boolean label depending on the string value ("true" -> True and "false" -> False).

Reviewer: ```suggestion
- In binary workspaces, the value `binary_label` is set as the label of the element for the category with ID `category_id`. If `binary_label == 'none'`, the element's label will be removed. Otherwise, the element will be assigned a boolean label depending on the string value ("true" -> True and "false" -> False).
```

> - In multiclass workspaces, the category to label is the one specified in the `category_id` parameter. If `category_id == 'none'`, the element's label will be removed.

Reviewer: ```suggestion
- In multiclass workspaces, the category to assign as the element's label is the one specified in the `category_id` parameter. If `category_id == 'none'`, the element's label will be removed.
```

**Example request:**

```json
{
   "mode": "MultiClass"
}
```

**Example response:**

```json
Stream of a csv file
```

:::
::::

### Export model

Download the trained model files for the given iteration index. If the iteration is not provided, the model from the latest iteration is exported. In order for this to work, the ModelType curently in use must implement the `ModelAPI.export_model()` method. In binary workspaces, the `category_id` specifies the selected category.

::::{tab-set}
:::{tab-item} Binary mode

<span class="request_type">GET</span> ```/workspace/<workspace_id>/export_model?category_id=<category_id>&iteration_index=<iteration_number>```

**Example request:**

```text
Empty
```

**Example response:**

```text
Stream of ZIP file
```

:::
:::{tab-item} Multiclass mode

<span class="request_type">GET</span> ```/workspace/<workspace_id>/export_model?iteration_index=<iteration_number>```

**Example request:**

```text
Empty
```

**Example response:**

```text
Stream of ZIP file
```

:::
::::

## Model Evaluation

Endpoints related to evaluation of a model.

### Get elements for precision evaluation

> Get a sample of elements that are predicted as positive for the category by the latest model. This sample can be used to evaluate the precision of the model predictions: the user should label this sample of elements. After they are labeled, an invocation of the ['Run precision evaluation'](#run-precision-evaluation) endpoint will compare the user-provided labels to the model predictions to estimate the precision of the latest model. This endpoint can be used only in binary workspaces.

Reviewer: ```suggestion
Get a sample of elements that are predicted as positive for the category by the latest model. This sample can be used to evaluate the precision of the model predictions: The user should label this sample of elements. After they are labeled, an invocation of the ['Run precision evaluation'](#run-precision-evaluation) endpoint will compare the user-provided labels to the model predictions to estimate the precision of the latest model. This endpoint can be used only in binary workspaces.
```

**Example request:**

```json
{
    "changed_elements_count": 20
}
```

**Example response:**

```json
{
    "canceled": "OK"
}
```

### Get elements for accuracy evaluation

Get a sample of predicted elements. This sample can be

> used to evaluate the accuracy of the model predictions: the user should label this sample of elements. After they are labeled, an invocation of the ['Run accuracy evaluation'](#run-accuracy-evaluation) endpoint will compare the user-provided labels to the model predictions to estimate the accuracy of the latest model. This endpoint can be used only in multiclass workspaces.

Reviewer: ```suggestion
used to evaluate the accuracy of the model predictions: The user should label this sample of elements. After they are labeled, an invocation of the ['Run accuracy evaluation'](#run-accuracy-evaluation) endpoint will compare the user-provided labels to the model predictions to estimate the accuracy of the latest model. This endpoint can be used only in multiclass workspaces.
```
